src/test/java/typewiseAlert/TypewiseAlertTest.java

```diff
@@ -0,0 +1,39 @@
+package typewiseAlert;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import batteryCharacter.AlertTarget;
+import batteryCharacter.BatteryCharacter;
+import batteryCharacter.BreachType;
+import batteryCharacter.CoolingType;
+import temparatureBreachType.TemparetureBreach;
+import temparatureBreachTypeImpl.TemparetureBreachImpl;
+import typewiseAlertImpl.TypewiseAlertImpl;
+
+public class TypewiseAlertTest {
+	public TemparetureBreach tempBreach = new TemparetureBreachImpl();
+	TypewiseAlert alert = new TypewiseAlertImpl();
+
+	@Test
+	public void infersBreachAsPerLimits() {
+		assertTrue("test", tempBreach.inferBreach(12, 20, 30) == BreachType.TOO_LOW);
+		assertTrue("test", tempBreach.inferBreach(22, 20, 30) == BreachType.NORMAL);
+		assertTrue("test", tempBreach.inferBreach(32, 20, 30) == BreachType.TOO_HIGH);
+
+	}
+	
+	@Test
+	public void veryfyBreachType() {
+		assertTrue("test", tempBreach.classifyTemperatureBreach(CoolingType.HI_ACTIVE_COOLING, 55) == BreachType.TOO_HIGH);
+		assertTrue("test", tempBreach.classifyTemperatureBreach(CoolingType.MED_ACTIVE_COOLING, 15) == BreachType.NORMAL);
+		assertTrue("test", tempBreach.classifyTemperatureBreach(CoolingType.MED_ACTIVE_COOLING, -10) == BreachType.TOO_LOW);
+	}
+	
```

Comment:
What if you were to test the content of the email / controller message, without having to actually integrate with a real inbox or hardware? How far can you prove your code?